lib/MySQL_Monitor.cpp

```diff
@@ -317,14 +317,16 @@ MYSQL * MySQL_Monitor_Connection_Pool::get_connection(char *hostname, int port,
 					break;
 				}
 #ifdef DEBUG
-				for (unsigned int j=0; j<conns->len; j++) {
-					MYSQL *my1 = (MYSQL *)conns->index(j);
-					assert(my!=my1);
-					assert(my->net.fd!=my1->net.fd);
-				}
+				if (my) {
```

Comment:
If my is NULL in debug, it will crash here 
assert(my->net.fd!=my1->net.fd);

Reply:
Previously, some work was done in get_connection to close connection if not used for long time. It can now return null if there are no open connections left in container.

Reply:
Comment added